pkg/pipelines/bootstrap.go

```diff
@@ -197,3 +220,40 @@ func marshalOutputs(out io.Writer, outputs []interface{}) error {
 	}
 	return nil
 }
+
+// validateImageRepo validates the input image repo.  It determines if it is
+// for internal registry and prepend internal registry hostname if neccessary.
+func validateImageRepo(o *BootstrapOptions) (bool, string, error) {
+	components := strings.Split(o.ImageRepo, "/")
+	errorMsg := "failed to parse image repo:%s, expected image repository in the form <registry>/<username>/<repository> or <project>/<app> for internal registry"
+
+	// repo url has minimum of 2 components
+	if len(components) < 2 {
+		return false, "", fmt.Errorf(errorMsg, o.ImageRepo)
+	}
+
+	// check for spaces
```

Comment:
It's not checking for spaces tho'?

It's ensuring that the paths aren't empty?

Reply:
Fixed comment.   Add new check for white spaces.  And test cases.